velox/type/Type.h

```diff
@@ -501,6 +533,79 @@ class TypeBase : public Type {
   }
 };
 
+/*
+ * A DecimalType tp re
+ */
+class DecimalType : public Type {
+ public:
+  DecimalType(int typmod = 4608)
+      : typmod_(typmod),
+        Type(
+            PRECISION(typmod) <= 18 ? TypeKind::DECIMAL
+                                    : TypeKind::DECIMAL128) {
+    isBigDecimal_ = !(PRECISION(typmod) <= 18);
+  }
+
+  inline bool isPrimitiveType() const override {
+    return isBigDecimal_ ? TypeTraits<TypeKind::DECIMAL128>::isPrimitiveType
+                         : TypeTraits<TypeKind::DECIMAL>::isPrimitiveType;
+  }
+
+  inline bool isFixedWidth() const override {
+    return isBigDecimal_ ? TypeTraits<TypeKind::DECIMAL128>::isFixedWidth
+                         : TypeTraits<TypeKind::DECIMAL>::isFixedWidth;
+  }
+
+  inline const char* kindName() const override {
+    return isBigDecimal_ ? TypeTraits<TypeKind::DECIMAL>::name
+                         : TypeTraits<TypeKind::DECIMAL128>::name;
+  }
+
+  inline uint32_t size() const override {
+    // BigDecimal TypeSignature can have 2 children which can be
```

Comment:
size() returns number of child nodes in the type tree; decimal type does not have any child types, hence, size() should return zero.